api/src/main/java/org/openmrs/module/m2sysbiometrics/bioplugin/NationalBioServerClient.java

```diff
@@ -24,6 +27,17 @@ protected Object getResponse(Object requestPayload) {
                 .marshalSendAndReceive(getServiceUrl(), requestPayload, message -> addAuthorizationHeader());
     }
 
+    public void handleRegistrationError(String nationalId, M2SysCaptureResponse capture) {
```

Comment:
This failure handling doesn't really belong in the national/local bioserver interface - the responsibility of this interface is to communicate through SOAP. This handling belong more a level above, in the registration service, eventually a separate interface.